modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -1039,75 +1039,115 @@ func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
+			ctx := suite.chainA.GetContext()
+			tc.malleate()
 
-	lastPacketSequence, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetCounterpartyLastPacketSequence(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
-	suite.Require().Equal(uint64(0), lastPacketSequence)
-	suite.Require().False(found)
+			if tc.expPanic {
+				suite.Require().Panics(func() {
+					suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.WriteUpgradeOpenChannel(ctx, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				})
+			} else {
+				suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.WriteUpgradeOpenChannel(ctx, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				channel := path.EndpointA.GetChannel()
 
-	events := ctx.EventManager().Events().ToABCIEvents()
-	expEvents := ibctesting.EventsMap{
-		types.EventTypeChannelUpgradeOpen: {
-			types.AttributeKeyPortID:                    path.EndpointA.ChannelConfig.PortID,
-			types.AttributeKeyChannelID:                 path.EndpointA.ChannelID,
-			types.AttributeCounterpartyPortID:           path.EndpointB.ChannelConfig.PortID,
-			types.AttributeCounterpartyChannelID:        path.EndpointB.ChannelID,
-			types.AttributeKeyChannelState:              types.OPEN.String(),
-			types.AttributeKeyUpgradeConnectionHops:     channel.ConnectionHops[0],
-			types.AttributeKeyUpgradeVersion:            channel.Version,
-			types.AttributeKeyUpgradeOrdering:           channel.Ordering.String(),
-			types.AttributeKeyUpgradeSequence:           fmt.Sprintf("%d", channel.UpgradeSequence),
-			types.AttributeKeyUpgradeChannelFlushStatus: channel.FlushStatus.String(),
-		},
-		sdk.EventTypeMessage: {
-			sdk.AttributeKeyModule: types.AttributeValueCategory,
-		},
-	}
-	ibctesting.AssertEvents(&suite.Suite, expEvents, events)
+				// Assert that channel state has been updated
+				suite.Require().Equal(types.OPEN, channel.State)
+				suite.Require().Equal(mock.UpgradeVersion, channel.Version)
+				suite.Require().Equal(types.ORDERED, channel.Ordering)
 
-	counterpartyUpgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetCounterpartyUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
-	suite.Require().Equal(types.Upgrade{}, counterpartyUpgrade)
-	suite.Require().False(found)
+				// Assert that state stored for upgrade has been deleted
+				upgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				suite.Require().Equal(types.Upgrade{}, upgrade)
+				suite.Require().False(found)
+
+				lastPacketSequence, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetCounterpartyLastPacketSequence(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				suite.Require().Equal(uint64(0), lastPacketSequence)
+				suite.Require().False(found)
```

Comment:
its okay to remove these lines, this state entry will be removed in an upcoming PR and we won't need to test this anymore :)